Tests iOS/MyList/ListenTests.swift

```diff
@@ -0,0 +1,115 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import XCTest
+import Sails
+
+class ListenTests: XCTestCase {
+    var server: Sails.Application!
+    var app: PocketAppElement!
+    var snowplowMicro = SnowplowMicro()
+
+    override func setUp() async throws {
+        continueAfterFailure = false
+
+        let uiApp = XCUIApplication()
+        app = PocketAppElement(app: uiApp)
+        await snowplowMicro.resetSnowplowEvents()
+
+        server = Application()
+
+        server.routes.post("/graphql") { request, _ -> Response in
+            return .fallbackResponses(apiRequest: ClientAPIRequest(request))
+        }
+
+        try server.start()
+    }
+
+    @MainActor
+    override func tearDown() async throws {
+        try server.stop()
+        app.terminate()
+        await snowplowMicro.assertNoBadEvents()
+    }
+
+    func test_listen_shows_whenInFlag() {
+        let flagsLoaded = expectation(description: "loaded flags")
+        server.routes.post("/graphql") { request, _ -> Response in
+            let apiRequest = ClientAPIRequest(request)
+            if apiRequest.isForFeatureFlags {
+                defer { flagsLoaded.fulfill() }
+                return .featureFlags("feature-flags-listen")
+            }
+
+            return .fallbackResponses(apiRequest: apiRequest)
+        }
+
+        app.launch().tabBar.savesButton.wait().tap()
+        app.saves.itemView(matching: "Item 1").wait()
+
+        wait(for: [flagsLoaded])
+
+        // do a refresh because the flag prob loaded in the background.
+        app.saves.pullToRefresh()
+
+        app.saves.filterButton(for: "Listen").wait().tap()
```

Comment:
Is there an expectation that's needed after this to test that Listen actually displayed? However, that'd rely on Listen having an accessibility identifier. Or, maybe there's a way of drilling down the view hierarchy for the right view? …though that's an ugly solution.

Reply:
I think I see it in a test below - `app.listenPlay.wait()` ?